Makefile

```diff
@@ -20,12 +20,21 @@ config.mk:
 	exit 1
 
 setup.bin: setup.ml
-	@ocamlopt.opt -o $@ $< || ocamlopt -o $@ $< || ocamlc -o $@ $<
-	@rm -f setup.cmx setup.cmi setup.o setup.cmo
+	@ocamlfind ocamlopt -o $@ -linkpkg -package oasis.dynrun setup.ml || ocamlfind ocamlc -o $@ -linkpkg -package oasis.dynrun setup.ml || true
+	@rm -f setup.cmi setup.cmo setup.cmx setup.o
 
 setup.data: setup.bin
 	@./setup.bin -configure $(ENABLE_TESTS) $(ENABLE_XEN) $(ENABLE_XENLIGHT) $(ENABLE_XENGUESTBIN) $(ENABLE_XENTOOLLOG)
 
+setup.ml: _oasis.in
+ifeq ($(BISECT_COVERAGE),YES)
+	rm -f _oasis
+	sed -e 's/BuildDepends:/BuildDepends: bisect_ppx,/' _oasis.in >_oasis
+else
+	ln -sf _oasis.in _oasis
+endif
+	oasis setup -setup-update dynamic
```

Comment:
I see we're doing oasis setup now instead of checking in the generated files into the repo, if I'm not mistaken?
But some generated files are still remaining, these also need to be removed.
You can run this command to remove all the files, and even the sections of files, that are generated by OASIS:
```
oasis setup-clean -replace-sections -remove
```